resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java

```diff
@@ -78,7 +78,8 @@ public static void writeNomapResponse(BuiltResponse jaxrsResponse, final HttpReq
       
       executeFilters(jaxrsResponse, request, response, providerFactory, method);
 
-      if (jaxrsResponse.getEntity() == null)
+      //[RESTEASY-1627] check on response.getOutputStream() to avoid resteasy-netty4 trying building a chunked response body for HEAD requests 
+      if (jaxrsResponse.getEntity() == null || response.getOutputStream() == null)

```

Comment:
I see, @asoldano. Thanks for the explanation. Ok, how about this: If verb is HEAD, commitHeaders() removes the Content-Length header. Let me say:

- I don't see anything wrong with the solution you've presented.
- I'm not trying to be difficult. I'm just wondering if there's a general solution that would apply to other server adaptors.